webapp/channels/src/components/admin_console/system_users/system_users_list/system_users_list.tsx

```diff
@@ -1,384 +1,266 @@
-        this.setState({
-            showEmailModal: true,
-            user,
-        });
-    };
-
-    doEmailResetDismiss = () => {
-        this.setState({
-            showEmailModal: false,
-            user: undefined,
-        });
-    };
-
-    doEmailResetSubmit = (user?: UserProfile) => {
-        if (user) {
-            this.props.actions.getUser(user.id);
-        }
-
-        this.setState({
-            showEmailModal: false,
-            user: undefined,
-        });
-    };
-
-    getInfoForUser(user: UserProfile) {
-        const info = [];
-
-        if (user.auth_service) {
-            let service;
-            if (user.auth_service === Constants.LDAP_SERVICE || user.auth_service === Constants.SAML_SERVICE) {
-                service = user.auth_service.toUpperCase();
-            } else {
-                service = Utils.toTitleCase(user.auth_service);
-            }
+enum ColumnNames {
+    displayName = 'displayNameColumn',
+    email = 'emailColumn',
+    createAt = 'createAtColumn',
+    lastLoginAt = 'lastLoginColumn',
+    lastStatusAt = 'lastStatusAtColumn',
+    lastPostDate = 'lastPostDateColumn',
+    daysActive = 'daysActiveColumn',
+    totalPosts = 'totalPostsColumn',
+    actions = 'actionsColumn',
+}
 
-            info.push(
-                <FormattedMarkdownMessage
-                    key='admin.user_item.authServiceNotEmail'
-                    id='admin.user_item.authServiceNotEmail'
-                    defaultMessage='**Sign-in Method:** {service}'
-                    values={{
-                        service,
-                    }}
-                />,
-            );
+const columnHelper = createColumnHelper<SystemUsersRow>();
+
+function SystemUsersList() {
+    const {formatMessage} = useIntl();
+
+    const currentUser = useSelector(getCurrentUser);
+
+    const tableId = 'systemUsersTable';
+
+    const columns = useMemo(
+        () => [
+            {
+                id: ColumnNames.displayName,
+                accessorKey: 'userDetails',
+                header: formatMessage({
+                    id: 'admin.system_users.list.userDetails',
+                    defaultMessage: 'User details',
+                }),
+                cell: (info: CellContext<SystemUsersRow, null>) => {
+                    return (
+                        <div>
+                            <div className='profilePictureContainer'>
+                                <img
+                                    className='profilePicture'
+                                    src={imageURLForUser(info.row.original.id)}
+                                    aria-hidden='true'
+                                />
+                            </div>
+                            <div
+                                className='displayName'
+                                title={info.row.original.display_name}
+                            >
+                                {info.row.original.display_name || ''}
```

Comment:
Do we need a blank string here? If `display_name` is `null` or `undefined` it would just show nothing right?

Reply:
Yes library expects the data if to be present if not them empty string.